Assets/Scripts/Snake/Snake.cs

```diff
@@ -137,14 +145,33 @@ private void UpdateSpline()
 
             DirectionOnCubeSide stepDirectionOnCubeSide = StepInputDirection.ToLocalDirectionOnCubeSide(ReferenceDirectionForInput);
             CubePoint nextPoint = GetPointOnCubeInDirection(stepDirectionOnCubeSide);
- 
+
             SplinePath.Spline.Add(CalculateSplineKnot(nextPoint));
-            SplinePath.Spline.RemoveAt(0);
-            
             Points.Add(nextPoint);
-            Points.RemoveAt(0);
-
-            UpdateSnakeBody();
+            
+            
+            if (removeNextPoint)
```

Comment:
rename "removeNextPoint" to smth that indicates that the snake is getting lager. 
E.g.: "shouldGrowNextUpdate"